lib/rust/ensogl/component/grid-view/src/lib.rs

```diff
@@ -323,23 +468,28 @@ impl<E: Entry> Model<E, E::Params> {
         text_layer: &Option<WeakLayer>,
     ) {
         use std::collections::hash_map::Entry::*;
-        let mut visible_entries = self.visible_entries.borrow_mut();
-        let mut free_entries = self.free_entries.borrow_mut();
         let create_new_entry = || {
             let text_layer = text_layer.as_ref().and_then(|l| l.upgrade());
             self.entry_creation_ctx.create_entry(text_layer.as_ref())
         };
-        let entry = match visible_entries.entry((row, col)) {
-            Occupied(entry) => entry.into_mut(),
-            Vacant(lack_of_entry) => {
-                let new_entry = free_entries.pop().unwrap_or_else(create_new_entry);
-                set_entry_position(&new_entry, row, col, entry_size);
-                new_entry.entry.frp().set_location((row, col));
-                self.display_object.add_child(&new_entry);
-                lack_of_entry.insert(new_entry)
-            }
+        let entry_frp = {
+            let mut visible_entries = self.visible_entries.borrow_mut();
+            let mut free_entries = self.free_entries.borrow_mut();
+            let entry = match visible_entries.entry((row, col)) {
+                Occupied(entry) => entry.into_mut(),
+                Vacant(lack_of_entry) => {
+                    let new_entry = free_entries.pop().unwrap_or_else(create_new_entry);
+                    set_entry_position(&new_entry, row, col, entry_size, &self.column_widths);
+                    new_entry.entry.frp().set_location((row, col));
+                    self.display_object.add_child(&new_entry);
+                    lack_of_entry.insert(new_entry)
+                }
+            };
+            entry.entry.frp().clone_ref()
```

Comment:
Good catch! There is still one FRP event emitted during borrow, in line 483. I wonder how to do that - maybe we can make this block returning entry_frp and a flag indicating if the entry was reassigned.

Also, add a comment why the block is needed.

Reply:
Fixed & added a comment